index.html

```diff
@@ -3330,43 +3413,81 @@ <h3>Algorithm</h3>
             <li>Otherwise, if <a>term definition</a> has a
               <a>type mapping</a>:
               <ol>
-                <li>Reference the value associated with the <code>@type</code>
-                  <a>entry</a> in <var>type/language map</var> using the variable
-                  <var>type map</var>.</li>
-                <li>If <var>type map</var> does not have a <a>entry</a> corresponding
+                <li>If <var>type map</var> does not have an <a>entry</a> corresponding
                   to the <a>type mapping</a> in <a>term definition</a>,
                   create one and set its value to the <a>term</a>
                   being processed.</li>
               </ol>
             </li>
+            <li>Reference the value associated with the <code>@language</code>
+              <a>entry</a> in <var>type/language map</var> using the variable
+              <var>language map</var>.</li>
+            <li class="changed">if <a>term definition</a> has both
+              a <a>language mapping</a> and a <a>direction mapping</a>:
+              <ol>
+                <li>Create a new variable <var>lang dir</var>.</li>
+                <li>If neither the <a>language mapping</a> and the <a>direction mapping</a>
+                  are `null`, set <var>lang dir</var> to the concatenation
+                  of <a>language mapping</a> and <a>direction mapping</a>
+                  separated by an underscore (`"_"`).</li>
+                <li>Otherwise, if <a>language mapping</a> is not `null`,
+                  set <var>lang dir</var> to the <a>language tag</a> in <a>language mapping</a>.
+                <li>Otherwise, if <a>direction mapping</a> is not `null`,
+                  set <var>lang dir</var> to <a>direction mapping</a>
+                  preceded by an underscore (`"_"`).</li>
```

Comment:
I guess I'm ok with this as a solution the language map + base direction issue. I'm not sure what else we'd do that wouldn't be much more complicated.

Reply:
It certainly threw a wrench in the value expansion/compaction and term selection logic :(. Fortunately, only minimal changes to the Compaction and Expansion algorithms, themselves.